mobile/test/common/BUILD

```diff
@@ -10,6 +10,7 @@ envoy_cc_test(
     repository = "@envoy",
     deps = [
         "//library/common:engine_common_lib",
+        "@envoy_build_config//:extension_registry",
```

Comment:
not OSS-related per se: we should be able to depend on it in our private envoy mobile wrapper repo and just call `registerFactories` from there too